### DISCUSSION TITLE
Restore an existing PVC

### DISCUSSION THREAD

**Discussion by fad3t** [vmware-tanzu/velero#6867] [Community support Q&A] [answered]
Hello,

I have a PVC that is being backed up on a daily basis.

Some data is corrupted on the volume, so I would like to restore it from the backup of yesterday.

From what I've read in the documentation, Velero won't replace existing resources if they're already present on the cluster, which is the case here. However, I tried using the `--existing-resource-policy=update` flag during the restore, hoping it would recreate the PVC (from the snapshot) but it did not.

Am I missing something? What would be the right way to do this (if that's even possible)?

Thanks,
Fred

**Comment by draghuram** [comment, marked as answer]
"update" doesn't replace data inside a PVC. See this note

"Update of a resource only applies to the Kubernetes resource data such as its spec. It may not work as expected for certain resource types such as PVCs and Pods. In case of PVCs for example, data in the PV is not restored or overwritten in any way."

at https://velero.io/docs/main/restore-reference/#restore-existing-resource-policy.

Some alternatives are to do a restore to an alternative namespace or to another cluster and then copy required files using network utilities such as scp, rsync etc. But I agree that it will be good for Velero to support this use case.

Thanks, Raghu (CloudCasa)

**Reply by fad3t** [reply]
OK, thanks for the feedback!

**Reply by sseago** [reply]
Another workaround would be to *delete* the PVC you need refreshed to its last backup state prior to restore. Note that if you're using fs backup, you'll also need to delete the mounting pod/deployment and restore those as well. If you're using snapshots, deleting the PVC should be sufficient.

**Reply by fad3t** [reply]
Indeed, that's what I've been doing for now. Thanks :)
